src/dvc_data/diff.py

```diff
@@ -69,16 +52,20 @@ def __bool__(self):
         return self.typ != UNCHANGED
 
 
-@dataclass
+@define
 class DiffResult:
-    added: List[Change] = field(default_factory=list, compare=True)
-    modified: List[Change] = field(default_factory=list, compare=True)
-    deleted: List[Change] = field(default_factory=list, compare=True)
-    unchanged: List[Change] = field(default_factory=list, compare=True)
+    added: List[Change] = field(factory=list, repr=reprlib.repr)
+    modified: List[Change] = field(factory=list, repr=reprlib.repr)
+    deleted: List[Change] = field(factory=list, repr=reprlib.repr)
+    unchanged: List[Change] = field(factory=list, repr=reprlib.repr)
```

Comment:
I added `reprlib.repr` so that the repr get size-limited/ truncated. Eg:
```python
DiffResult(added=[], modified=[Change(old=Tr... typ='modify')], deleted=[Change(old=Tr... typ='delete'), Change(old=Tr... typ='delete'), Change(old=Tr... typ='delete'), Change(old=Tr... typ='delete'), Change(old=Tr... typ='delete'), Change(old=Tr... typ='delete'), ...], unchanged=[Change(old=Tr...p='unchanged'), Change(old=Tr...p='unchanged'), Change(old=Tr...p='unchanged'), Change(old=Tr...p='unchanged'), Change(old=Tr...p='unchanged'), Change(old=Tr...p='unchanged'), ...])
```